openstf/model/split_energy.py

```diff
@@ -42,23 +44,65 @@ def split_energy(pid):
     error = components[["load", "Inschatting"]].diff(axis=1).iloc[:, 1]
     mae = error.abs().mean()
     coefdict.update({"MAE": mae})
+    coefsdf = convert_coefdict_to_coefsdf(pj, input_split_function, coefdict)
 
-    # Get average coefs of previous runs
+    # Get average coefs of previous runs and check if new coefs are valid
     mean_coefs = db.get_energy_split_coefs(pj, mean=True)
-    # Loop over keys and check if the difference with the average value is not more than 100%
-    # In case the difference is more tha 100% of the average set KPI coefs as expected to False
+    if are_new_coefs_valid(coefdict, mean_coefs) is False:
+        # If coefs not valid, do not update the coefs in the db and send teams
+        # message that something strange is happening
+        monitoring.post_teams_alert(
+            "New splitting coefficients for pid {} deviate strongly from previously stored coefficients.".format(
+                pj["id"],
+            ),
```

Comment:
Would make sense to include some information on which coefs are considered incorrect, or at least how the new coefs compare to the old onse

Reply:
I stop checking when one key is failing and added info of that key, new key value and old key value.

Reply:
After a comment of JM, I changed it to checking for all keys and giving back all failing keys